pkg/logs/input/file/scanner.go

```diff
@@ -185,23 +186,50 @@ func (s *Scanner) launchTailers(source *config.LogSource) {
 		if _, isTailed := s.tailers[file.Path]; isTailed {
 			continue
 		}
-		var tailFromBeginning bool
+
+		mode, found := TailingModeFromString(source.Config.TailingMode)
+		if !found {
+			log.Infof("Invalid or no tailing mode provided, tailing from the end of the file %v", file.Path)
+			source.Config.TailingMode = mode.String()
+		}
+		if wildcard && (mode == Beginning || mode == ForceBeginning) {
+			log.Warnf("Tailing from the beginning is not supported for wildcard path %v, tailing from the end", file.Path)
+			source.Config.TailingMode, mode = TailingMode(End).String(), End
+		}
+
 		if source.Config.Identifier != "" {
 			// only sources generated from a service discovery will contain a config identifier,
 			// in which case we want to collect all logs.
 			// FIXME: better detect a source that has been generated from a service discovery.
-			tailFromBeginning = true
+			mode = Beginning
 		}
-		s.startNewTailer(file, tailFromBeginning)
+		s.startNewTailer(file, mode)
 	}
 }
 
 // startNewTailer creates a new tailer, making it tail from the last committed offset, the beginning or the end of the file,
 // returns true if the operation succeeded, false otherwise
-func (s *Scanner) startNewTailer(file *File, tailFromBeginning bool) bool {
+func (s *Scanner) startNewTailer(file *File, m TailingMode) bool {
 	tailer := s.createTailer(file, s.pipelineProvider.NextPipelineChan())
 
-	offset, whence, err := Position(s.registry, tailer.Identifier(), tailFromBeginning)
+	var offset int64
+	var whence int
+	var mode TailingMode
+	previousMode, _ := TailingModeFromString(s.registry.GetTailingMode(tailer.Identifier()))
+	if previousMode != m {
```

Comment:
Can you explain why we need a specific logic for this case please ? Maybe with a comment ? Also it would be nice to move all this logic to a dedicated method to make the code more readable

Reply:
I agree, an explanation is definitely needed. The idea is to have a clearly defined behaviour in case of tailing mode change. That being said I should probably add that to the documentation as well.